### PR TITLE
[Attention][Feature] Implement compute-communication overlap for pure-prefill batches

### PR DESCRIPTION
**Pull request by qjgggsse**
What this PR does / why we need it?

  This PR implements a delayed allgather optimization for pure-prefill batches in DeepSeek V4 DSA attention. Each    
  rank computes wq_a/wkv on its local SP partition first, then allgathers smaller intermediates instead of the full  
  hidden_states. Communication and computation are overlapped via dual-stream (main + aux) scheduling.

  Does this PR introduce any user-facing change?

  Yes, adds a new configuration parameter prefill_comm_compute_overlap (default: disabled).

  How was this patch tested?

  Verified correctness (output logits match baseline) on DeepSeek V4 with TP+SP, pure prefill and PD-disaggregated   
  prefill scenarios.

- vLLM version: v0.20.2
- vLLM main: https://github.com/vllm-project/vllm/commit/1ac10f159a09897baada01b14b6a0dd6442aefd6

### CODE REVIEW

File: vllm_ascend/attention/dsa_v1.py [hunk 1]
```diff
@@ -1623,12 +1646,91 @@ def _mla_prolog_multistream(self, hidden_states, cos, sin, swa_kv_cache, slot_ma
...
+                -1, (self.n_local_heads, self.head_dim))
+            q = triton_q_rms(q, self.eps)
+
+        kv = torch.ops.vllm.maybe_all_gather_and_maybe_unpad(kv, True)
+        kv = kv[:num_actual_tokens]
+        if self.compress_ratio > 1:
+            full_hidden_states = (
+                torch.ops.vllm.maybe_all_gather_and_maybe_unpad(
+                    hidden_states, True))
+            full_hidden_states = full_hidden_states[:num_actual_tokens]
```
[the remark this answers is not in the document]
Comment (qjgggsse) [reply vllm_ascend/attention/dsa_v1.py:1731]: The allgather + full scatter pattern for KV is inherited from the baseline serial path (line 1793-1807). This PR 
  focuses on overlapping the existing computation and communication, not restructuring the KV scatter logic.
  Optimizing to local-only scatter would require changes to the upstream slot_mapping and cos/sin passing, which is  
  out of scope for this PR. Filed as a follow-up optimization.

File: vllm_ascend/attention/dsa_v1.py [hunk 2]
```diff
@@ -1623,12 +1646,91 @@ def _mla_prolog_multistream(self, hidden_states, cos, sin, swa_kv_cache, slot_ma
...
+
+        with npu_stream_switch(aux_stream, enabled=True):
+            torch.npu.current_stream().wait_event(e_phase3)
+            q_b_quant, q_b_scale = self.cv_wq_b.quantize(qr)
+            q = self.cv_wq_b.matmul(q_b_quant, q_b_scale).unflatten(
+                -1, (self.n_local_heads, self.head_dim))
+            q = triton_q_rms(q, self.eps)
+
+        kv = torch.ops.vllm.maybe_all_gather_and_maybe_unpad(kv, True)
+        kv = kv[:num_actual_tokens]
```
[the remark this answers is not in the document]
Comment (qjgggsse) [reply vllm_ascend/attention/dsa_v1.py:1728]: The allgather + full scatter pattern for KV is inherited from the baseline serial path. This PR focuses on         overlapping existing computation and communication without restructuring the KV scatter logic. Optimizing to 
  local-only scatter requires upstream changes to slot_mapping and cos/sin passing, which is out of scope here.
